graphql_compiler/query_formatting/common.py

```diff
@@ -34,37 +35,18 @@
 ######
 
 
-def _raise_invalid_type_error(name, expected_python_type_name, value):
+def _raise_invalid_type_error(
+    name: str, expected_python_types: Collection[Type], value: Any
+) -> None:
     """Raise a GraphQLInvalidArgumentError that states that the argument type is invalid."""
     raise GraphQLInvalidArgumentError(
-        "Invalid type for argument {}. Expected {}. Got value {} of "
-        "type {}.".format(name, expected_python_type_name, value, type(value).__name__)
+        f"Invalid type for argument {name}. Expected one of {expected_python_types}. Got value "
+        f"{value} of type {type(value).__name__} instead."
     )
 
 
-def _deserialize_anonymous_json_argument(expected_type: GraphQLScalarType, value: Any) -> Any:
-    """Deserialize argument. See docstring of deserialize_json_argument.
-
-    Args:
-        expected_type: GraphQL type we expect.
-        value: object that can be interpreted as being of that type
-
-    Returns:
-        a value of the type produced by the parser of the expected type:
-            GraphQLDate: datetime.date
-            GraphQLDateTime: datetime.datetime with tzinfo=pytz.utc
-            GraphQLFloat: float
-            GraphQLDecimal: decimal.Decimal
-            GraphQLInt: int
-            GraphQLString: str
-            GraphQLBoolean: bool
-            GraphQLID: str
-
-    Raises:
-        ValueError if the value is not appropriate for the type. ValueError is chosen because
-        it is already the base case of exceptions raised by the GraphQL parsers.
-    """
-    allowed_types_for_graphql_type = {
+_ALLOWED_JSON_SCALAR_TYPES = MappingProxyType(
```

Comment:
Raymond Hettinger recommended wrapping a anonymous python dict with the `MappingProxyType` as the easiest way of making dictionaries immutable. He is a one of the core python contributors.

Reply:
In general, I agree with this suggestion, but I don't think that in this specific scenario it makes sense to include the comment.

I don't think that I should just include the comment in this specific place. The commenting methodology would be inconsistent. (We sometimes add a comment explaining why we use `MappingProxyType` and sometimes we don't). 

I also don't think that I should include the comment every single time that I use the type. That would clutter the codebase and I think that the `MappingProxyType` docstring is already pretty well written.